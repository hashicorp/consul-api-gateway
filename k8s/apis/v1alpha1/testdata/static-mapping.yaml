apiVersion: consul-api-gateway.hashicorp.com/v1alpha1
kind: GatewayClassConfig
metadata:
  name: test-gateway-class-config
spec:
  logLevel: info
  consul:
    authentication:
      account: consul-api-gateway
      method: consul-api-gateway
    address:
      host.docker.internal
    ports:
      http: 443
  useHostPorts: true
  images:
    consul-api-gateway: "consul-api-gateway:1"
    envoy: "envoy:1"
  nodeSelector:
    "ingress-ready": "true"
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: GatewayClass
metadata:
  name: test-gateway-class
spec:
  controller: "hashicorp.com/consul-api-gateway-gateway-controller"
  parametersRef:
    group: consul-api-gateway.hashicorp.com
    kind: GatewayClassConfig
    name: test-gateway-class-config
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: test-static-mapping
spec:
  gatewayClassName: test-gateway-class
  listeners:
  - protocol: HTTP
    port: 8083
    name: http
    allowedRoutes:
      namespaces:
        from: Same
  - protocol: HTTPS
    port: 8443
    name: https
    allowedRoutes:
      namespaces:
        from: Same
