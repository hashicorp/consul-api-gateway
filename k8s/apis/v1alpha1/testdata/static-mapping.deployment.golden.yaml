metadata:
  creationTimestamp: null
  labels:
    managedBy: consul-api-gateway
    name: test-static-mapping
  name: test-static-mapping
spec:
  selector:
    matchLabels:
      managedBy: consul-api-gateway
      name: test-static-mapping
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        managedBy: consul-api-gateway
        name: test-static-mapping
    spec:
      containers:
      - command:
        - /bootstrap/consul-api-gateway
        - exec
        - -log-json
        - -log-level
        - info
        - -gateway-host
        - $(IP)
        - -gateway-name
        - test-static-mapping
        - -consul-http-address
        - host.docker.internal
        - -consul-http-port
        - "443"
        - -consul-xds-port
        - "8502"
        - -envoy-bootstrap-path
        - /bootstrap/envoy.json
        - -envoy-sds-address
        - consul-api-gateway-controller.default.svc.cluster.local
        - -envoy-sds-port
        - "9090"
        - -acl-auth-method
        - consul-api-gateway
        env:
        - name: IP
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
        - name: HOST_IP
          valueFrom:
            fieldRef:
              fieldPath: status.hostIP
        image: envoyproxy/envoy:v1.19-latest
        name: consul-api-gateway
        ports:
        - containerPort: 20000
          name: ready
          protocol: TCP
        - containerPort: 8083
          hostPort: 8083
          name: http
          protocol: TCP
        - containerPort: 8443
          hostPort: 8443
          name: https
          protocol: TCP
        readinessProbe:
          httpGet:
            path: /ready
            port: 20000
        resources: {}
        volumeMounts:
        - mountPath: /bootstrap
          name: bootstrap
        - mountPath: /certs
          name: certs
      initContainers:
      - command:
        - cp
        - /bin/consul-api-gateway
        - /bootstrap/consul-api-gateway
        image: hashicorp/consul-api-gateway:0.1.0
        name: consul-api-gateway-init
        resources: {}
        volumeMounts:
        - mountPath: /bootstrap
          name: bootstrap
        - mountPath: /certs
          name: certs
      nodeSelector:
        ingress-ready: "true"
      serviceAccountName: consul-api-gateway
      volumes:
      - emptyDir: {}
        name: bootstrap
      - emptyDir: {}
        name: certs
status: {}
